components/centraldashboard/public/components/main-page.js

```diff
@@ -27,88 +28,130 @@ import '@polymer/neon-animation/animations/fade-out-animation.js';
-                    {text: 'Share solution on AI Hub', icon: 'donut-large'},
-                ],
-            },
-            quickLinks: {
-                type: Array,
-                value: [
                     {
-                        text: 'Open docs',
-                        link: 'https://www.kubeflow.org/docs/started/getting-started/',
+                        iframeUrl: '/jupyter/',
+                        text: 'Notebooks',
+                        href: '/notebooks',
                     },
                     {
-                        text: 'Open Github',
-                        link: 'https://github.com/kubeflow/kubeflow',
+                        iframeUrl: '/tfjobs/ui/',
+                        text: 'TFJob Dashboard',
+                        href: '/tjob-dashboard',
+                    },
+                    {
+                        iframeUrl: '/katib/',
+                        text: 'Katib Dashboard',
+                        href: '/katib-dashboard',
+                    },
+                    {
+                        iframeUrl: '/pipeline/',
+                        text: 'Pipeline Dashboard',
+                        href: '/pipeline-dashboard',
                     },
                 ],
             },
+            hideToolbar: {type: Boolean, value: false},
             sidebarItemIndex: {type: Number, value: 0},
-            primaryViewIndex: {type: Number, value: 0},
-            homeOrIframeViewIndex: {type: Number, value: 0},
-            url: {type: String, value: ''},
+            iframeUrl: {type: String, value: ''},
             buildVersion: {type: String, value: '0.4.1'},
             dashVersion: {type: String, value: VERSION},
-            _devMode: {type: Boolean, value: false},
+            _devMode: {type: Boolean, value: DEVMODE},
         };
     }
 
-    openExternalLink(href) {
-        const a = document.createElement('a');
-        a.href = href;
-        a.target = '_blank';
-        a.click();
+    /**
+      * Array of strings describing multi-property observer methods and their
+      * dependant properties
+      */
+    static get observers() {
+        return [
+            '_routePageChanged(routeData.page)',
+        ];
     }
 
-    openQuickLink(e) {
-        const {link} = e.model.item;
-        this.openExternalLink(link);
-    }
-
-    openLink(e) {
-        const {link, defaultPage} = e.model.item;
-        this.homeOrIframeViewIndex = defaultPage ? 0 : 1;
-        if (defaultPage) return;
-        this.url = link;
+    /**
+     * Intercepts any external links and ensures that they are captured in
+     * the route and sent to the iframe source.
+     * @param {MouseEvent} e
+     */
+    openInIframe(e) {
+        const url = new URL(e.currentTarget.href);
+        window.history.pushState({}, null, `_${url.pathname}`);
+        window.dispatchEvent(new CustomEvent('location-changed'));
+        e.preventDefault();
```

Comment:
Does this actually work in the cluster? If it's a matter of unique routes, we can prefix all our routes with `d/` for `dashboard`.

Reply:
It does. I also considered the anchor links approach which can be configured in [app-location](https://www.webcomponents.org/element/PolymerElements/app-route/elements/app-location#property-useHashAsPath). That's a more futureproof solution if we are concerned about collisions.